kore-rpc-types/src/Kore/JsonRpc/Types.hs

```diff
@@ -107,8 +107,10 @@ instance FromRequest (API 'Req) where
 
 data ExecuteState = ExecuteState
     { term :: KoreJson
-    , substitution :: Maybe KoreJson
     , predicate :: Maybe KoreJson
+    , ruleSubstitution :: Maybe KoreJson
+    , rulePredicate :: Maybe KoreJson
+    , ruleId :: Maybe Text
```

Comment:
Could we instead just modify the `nextStates` field to contain the rule information paired with the result states?

`nextStates :: Maybe [(RuleInfo, ExecuteState)]`

Reply:
I think what we need to do is to have indeed have a `RuleInfo` datatype as @jberthold suggests below , but store it as a field of `ExecuteState`. Moreover, perhaps we should store a list of `RuleInfo`: this way we can accommodate straight-line execution and return a list of rules that lead to a certain state.

Reply:
> nextStates :: Maybe [(RuleInfo, ExecuteState)]

This would be a breaking change, so I don't think we should go down that route. We could go with @geo2a's suggestion of adding it as a field in `ExecuteState`, but honestly it seems cosmetic whether we package them up in a separate object or just keep them as fields of `ExecuteState`.

Reply:
We agreed in the daily meeting to not introduce braking changes just yet.